compiler/rustc_typeck/src/check/wfcheck.rs

```diff
@@ -194,12 +194,13 @@ pub fn check_trait_item(tcx: TyCtxt<'_>, def_id: LocalDefId) {
     let hir_id = tcx.hir().local_def_id_to_hir_id(def_id);
     let trait_item = tcx.hir().expect_trait_item(hir_id);
 
-    let method_sig = match trait_item.kind {
-        hir::TraitItemKind::Fn(ref sig, _) => Some(sig),
-        _ => None,
+    let (method_sig, ty) = match trait_item.kind {
```

Comment:
Why not return selectively use eitehr `trait_item.span` or `ty.span` here? Then you don't need to pass through `ty`?